Refac: Add generic process status check

murilommen opened the pull request:
## Description

This repository needs to make the `status` method that is declared on ProcessRollingLogger to be placed 1 structure above, so that any ProcessActor might benefit from it.

- [x] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

Commit:
Switch the pipe signaler to use faster-fifo

This was part of perf optimization work to get batch validation working
faster in the process logger. It needs more work on the validation side
in whylogs in order to fully reap the benefits but its still faster to
use then the older pipe signlaer that was based on mp.Pipe.

The purpose of the signaler is to bridge the asynchronous actor messages
to their sender so they can mimic synchronous invocations. They use a
rendezvous strategy where the sender creates a Future that the pipe
signaler holds and waits for the logger process to signal its
completion.

murilommen committed:
restructuring code - partially working

Commit:
Create a protocol for a syncable message

murilommen committed:
making status go 1 level upwards

murilommen committed:
Merge branch 'dev/murilommen/generic-process-health-checks' of github.com:whylabs/whylogs into dev/murilommen/generic-process-health-checks

murilommen committed:
make threading imports

murilommen committed:
changes to suit for inheritance

murilommen committed:
Merge branch 'mainline' into dev/murilommen/generic-process-health-checks

murilommen committed:
bump back version

murilommen committed:
type fixes

murilommen committed:
type fixes

murilommen committed:
Merge branch 'dev/murilommen/generic-process-health-checks' of github.com:whylabs/whylogs into dev/murilommen/generic-process-health-checks

murilommen committed:
black fixes

murilommen committed:
pre-commit fixes

murilommen committed:
make PipeSignaler of type Any

naddeoa commented on python/whylogs/api/logger/experimental/logger/actor/pipe_signaler.py:
```diff
@@ -0,0 +1,92 @@
+        """
+        ... [Your existing docstring] ...
+        """
+        self.queue.send(result)
```
these doc strings were removed by chat gpt. Can you sub them back in from the old code.

naddeoa commented on python/whylogs/api/logger/experimental/logger/actor/process_rolling_logger.py:
```diff
@@ -344,28 +323,21 @@ def process_log_messages(self, messages: List[LogMessage]) -> None:
             self.process_log_dicts(log_dicts)
-            for message in messages:
-                self._signal(message.id, None)
+            # for message in messages:
```
can delete

naddeoa commented on python/whylogs/api/logger/experimental/logger/actor/process_rolling_logger_messages.py:
```diff
@@ -180,7 +194,9 @@ class PubSubEmbeddingDict(TypedDict):
-def _decode_pubsub_data(data: Union[PubSubEmbeddingDict, PubSubDict]) -> Optional[bytes]:
+def _decode_pubsub_data(
+    data: Union[PubSubEmbeddingDict, PubSubDict],
+) -> Optional[bytes]:
```
This didn't actually change right? Looks like it shouldn't have been reformatted either based on the length.

naddeoa reviewed (commented):
I'll check things out and inspect them locally soon too to make sure everything looks right

murilommen committed:
Rewrite docstrings on PipeSignaler and remove comments

murilommen committed:
revert unwanted line breaks

murilommen committed:
Merge branch 'mainline' into dev/murilommen/generic-process-health-checks

Commit:
Fix type issue in pipe signaler

This updates the generic type arguments of the pipe signaler to fix the
type error.

naddeoa commented on python/whylogs/api/logger/experimental/logger/actor/process_actor.py:
```diff
@@ -45,6 +57,13 @@ def __init__(
+        # but BaseProcessRollingLogger appears to need some different things to work
+        # properly
+        self._pipe_signaler: Optional[PipeSignaler[Any]] = PipeSignaler() if self._sync_enabled is True else None
+
```
Updated this branch with the fix
